pilot/pkg/kube/inject/inject.go

```diff
@@ -744,18 +758,22 @@ func intoObject(sidecarTemplate string, valuesConfig string, meshconfig *meshcon
 	switch v := out.(type) {
 	case *v2alpha1.CronJob:
 		job := v
+		typeMeta = &job.TypeMeta
 		metadata = &job.Spec.JobTemplate.ObjectMeta
 		deploymentMetadata = &job.ObjectMeta
 		podSpec = &job.Spec.JobTemplate.Spec.Template.Spec
 	case *corev1.Pod:
 		pod := v
+		typeMeta = &pod.TypeMeta
 		metadata = &pod.ObjectMeta
 		deploymentMetadata = &pod.ObjectMeta
 		podSpec = &pod.Spec
 	default:
 		// `in` is a pointer to an Object. Dereference it.
 		outValue := reflect.ValueOf(out).Elem()
```

Comment:
please add `corev1.Deployment`  as additional case, with perhaps a fallthru to make this clearer.

But otherwise it looks fine, better you, list the other objects that you we care about

Reply:
Done. See above.